yext/answers-headless-react
# Update React Sample App according to new Headless State

Update React sample app and tests based on the updated headless state

- update components to use the new vertical and universal state interfaces, and the new `SearchStatusState` and `QueryState`
- update `SearchBar` to manage its own autocomplete results in component state. Added a useAutocomplete hook
  - Note: autocomplete dropdown will flicker between old and recently trigger search results. This will be address through supporting async and non-async in `inputDropdown` in another item
- update `AppliedFilter` and `StaticFilter` to no longer use ID with headless public interface
- as discussed with product, any update to static Filter should reset facet to avoid invalid facet send as part of request. Added call to `resetFacets` on `handleFilterOptionChange`


J=SLAP-1693
TEST=manual

jest tests passed
smoke tested sample-app with new headless version
- see that universal search executed properly
- see that vertical search work with multiple static filters
- see that UI for loading state behave properly with universal/vertical search
- see that autocomplete display options properly
